app/controllers/provider/admin/messages/outbox_controller.rb

```diff
@@ -64,10 +67,14 @@ def recipient_ids
   end
 
   def mass_message?
-    current_account.provider? && params[:to].blank?
+    current_account.provider? && permitted_params[:to].blank?
   end
 
   def message_params
-    params.fetch(:message, {}).merge(:origin => "web")
+    permitted_params.fetch(:message, {}).merge(:origin => "web")
```

Comment:
I believe we don't need to permit `origin` if we are overwriting it here.

Reply:
You're right. In fact, I think there shouldn't be a general permitted params but instead each param should be retrieved separately